### PR TITLE
Fixed bugs in + and == operators in Text

### PR DESCRIPTION
**Pull request by tomasmahlberg**
Issue #785.

### CODE REVIEW

File: source/base/Text.ooc [hunk 1]
```diff
@@ -51,7 +51,7 @@ Text: cover {
 	operator == (string: String) -> Bool { this == This new(string) }
 	operator == (other: This) -> Bool {
 		result := this _buffer == other _buffer
-		if (this _buffer raw != other _buffer raw)
```
Comment (davidhesselbom) [inline source/base/Text.ooc]: Haha, oops...

**Comment by davidhesselbom** [comment]
@marcusnaslund Can you try to give this priority? It'd be awfully nice to get rid of that invalid free before people learn to ignore that sort of error.

**Comment by marcusnaslund** [comment]
Yep, coming up.

**Comment by simonmika** [comment]
@tomasmahlberg Shouldn't this be a part of `TextBuffer` instead?

**Comment by tomasmahlberg** [comment]
@simonmika Might make more sense actually, since the operators work mostly with the `TextBuffer`s anyway. What do you think?

**Comment by simonmika** [comment]
Do a PR.

**Pull request merged**